### PR TITLE
Add support for metrics

### LINKED ISSUE #24: Add support for measurements
Hi, it would be great to have a support for measurements (as per https://opencensus.io/stats/measurement/) which is supported in Erlang library already (https://hexdocs.pm/opencensus/oc_stat_measure.html). I would be happy to contribute it if it's something you would like to have in the library.

### PR DESCRIPTION
**Pull request by mszczygiel**
fixes https://github.com/opencensus-beam/opencensus_elixir/issues/24

### CODE REVIEW

File: lib/opencensus/metrics.ex [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  Metrics.record(:name, %{tag1: "v1", tag2: "v2"}, 3)
+  Metrics.record([another: 1, name: 100], %{tag1: "v1", tag2: "v2})
+  ```
+  or using tag values that are present in process dictionary:
+  ```elixir
+  Metrics.record(:name, 3)
+  ```
+  """
+
+  def new_measure(name, description, unit) do
```
Comment (hauleth) [inline lib/opencensus/metrics.ex]: Why not just `new`?

File: lib/opencensus/metrics.ex [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    :oc_stat_measure.new(name, description, unit)
```
Comment (hauleth) [inline lib/opencensus/metrics.ex]: I would prefer to use `defdelegate` instead of "handcrafted" function.

File: lib/opencensus/metrics.ex [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      description,
+      tags,
+      {:oc_stat_aggregation_distribution, buckets: buckets}
+    )
+  end
+
+  @doc """
+  records single measure
+  """
+  def record(measure, %{} = tags, value) when is_number(value) do
```
Comment (hauleth) [inline lib/opencensus/metrics.ex]: I would make `tags` 3rd argument and I would accept keyword list as well to simplify usage.

File: config/config.exs [hunk 4]
```diff
@@ -8,6 +8,14 @@ if Mix.env() == :test do
...
-    reporters: [{Opencensus.TestSupport.SpanCaptureReporter, []}],
-    send_interval_ms: 100
+    reporters: [
+      {Opencensus.TestSupport.SpanCaptureReporter, []}
+    ],
+    send_interval_ms: 100,
+    stat: [
+      export_interval: 100,
+      exporters: [
+        {Opencensus.TestSupport.MetricsCaptureExporter, []}
```
Comment (hauleth) [inline config/config.exs:18]: I would dynamically attach exporter in tests as it will allow to make test async.

**Review by hauleth — COMMENTED** [review]
While this is nice, we currently focusing on getting on track with OpenTelemetry and few thing can change till then.

File: lib/opencensus/metrics.ex (hunk 2 above)
Comment (mszczygiel) [reply lib/opencensus/metrics.ex]: I did not know about `defdelegate`. Thanks!

File: config/config.exs (hunk 4 above)
Comment (mszczygiel) [reply config/config.exs:18]: Hmm, I don't think attaching it dynamically makes any difference. I think it's fine to just make tests async (which I just did)

File: lib/opencensus/metrics.ex (hunk 3 above)
Comment (mszczygiel) [reply lib/opencensus/metrics.ex]: sure

File: lib/opencensus/metrics.ex (hunk 1 above)
Comment (mszczygiel) [reply lib/opencensus/metrics.ex]: Yeah, `new` looks better.

**Pull request closed without merging**